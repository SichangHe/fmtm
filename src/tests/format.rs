use super::*;

fn default_format(input: &str) -> Result<String, std::fmt::Error> {
    format(input, Some(80))
}

fn format_20(input: &str) -> Result<String, std::fmt::Error> {
    format(input, Some(20))
}

fn format_inf(input: &str) -> Result<String, std::fmt::Error> {
    format(input, None)
}

macro_rules! t {
    ($name:ident, $input:expr) => {
        #[test]
        fn $name() {
            init_tracing();
            let input = $input.trim_start();
            let default_formatted = default_format(input).unwrap();
            assert_snapshot!(default_formatted);
            let formatted_20 = format_20(input).unwrap();
            assert_snapshot!(formatted_20);
            let formatted_inf = format_inf(input).unwrap();
            assert_snapshot!(formatted_inf);
        }
    };
}

t!(gpt_inline, include_str!("gpt_inline.md"));

t!(gpt1, include_str!("gpt1.md"));
