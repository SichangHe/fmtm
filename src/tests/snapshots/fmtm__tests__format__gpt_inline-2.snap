---
source: src/tests/format.rs
expression: formatted_20
---
# Short Markdown Example

Lorem ipsum dolor
sit amet,
**consectetur**
adipiscing *elit*.
Integer nec odio.
Praesent
**libero**.

## Subheading Level 2

*Sed cursus* ante
*dapibus* diam.
Sed **nisi**.
Nulla quis +sem+ at
-nibh- elementum
~imperdiet~.
**Duis** sagittis
*ipsum*.
Praesent
**mauris**.

- **Bold** with asterisks: ***strong emphasis***
- *Italic* with underscores: **italic emphasis**
- **Combined** styles: ***bold and italic***

### Subheading Level 3

**In markdown,
you can have:**

1. **Ordered lists**
    1. *Nested item 1*
    1. *Nested item 2*
1. **Unordered lists**
    - +List+ item
        with a plus
    - -List- item
        with
        a minus
    - *List* item with an asterisk

> "Blockquote with
> **bold** and
> *italic* text." -
> Famous Quote

```javascript
// Sample code block
function greet() {
    console.log("Hello, Markdown!");
}
```

And here's some
`inline code`
within a sentence.

**Thanks for
reading!**
